components/email/src/Email.svelte

```diff
@@ -780,30 +780,50 @@
     currentTooltipId = event.detail.tooltipID;
   }
 
-  function showFromParticipants(
-    messages: Message[],
-    participants: Participant[],
-  ) {
-    let fromParticipants = messages[messages.length - 1]?.from;
-    const hasFirstFromParticipant =
-      messages &&
-      participants &&
-      messages.length > 0 &&
-      fromParticipants.length;
-    const hasMoreParticipants = showSecondFromParticipant(
-      messages,
-      participants,
-    );
-    let fromString = "";
-    if (hasFirstFromParticipant) {
-      fromString += fromParticipants[0].name || fromParticipants[0].email;
-      if (hasMoreParticipants) {
-        fromString += ", " + (participants[0].name || participants[0].email);
+  function getParticipants(thread: Thread): string[] {
+    const messages = thread.messages;
+    const drafts = thread.drafts;
+    const lastMessageFrom = messages[messages.length - 1]?.from;
+    const hasParticipant =
+      messages && participants && messages.length > 0 && lastMessageFrom.length;
+
+    let participantsList = [];
+    if (drafts.length) {
+      participantsList.push("Draft");
+    }
+    if (hasParticipant) {
+      const participantName =
+        lastMessageFrom[0].email === userEmail
+          ? "Me"
+          : lastMessageFrom[0].name || lastMessageFrom[0].email;
+      participantsList.push(participantName);
+
+      //Only display 2 participants
+      const secondParticipant = messages
+        .slice(0, -1)
+        .reverse()
+        .find((msg) => {
+          return (
+            msg.from?.length > 0 &&
+            msg.from[0].email !== lastMessageFrom[0].email
+          );
+        });
+
+      if (secondParticipant) {
+        const secondParticipantName =
+          secondParticipant.from[0].email === userEmail
+            ? "Me"
+            : secondParticipant.from[0].name || secondParticipant.from[0].email;
+        participantsList.push(secondParticipantName);
       }
```

Comment:
@yifanplanet wouldn't this have been a lot simpler if you looped through the first two participants? 

If it's a draft, you can push `Draft` during the loop, else, do a check for the `me` and then end if we're at max 2 participants?

Reply:
Hi @AaronDDM Thanks for the review! It is true that looping through the messages would be way simpler. 

My original refactor was only to replace the `showSecondFromParticipant` function in here while keeping all other codes the same.

And we can't loop through the `participants` array because it's not ordered with message sending timestamp.